app/src/main/assets/test.html

```diff
@@ -33,76 +26,146 @@
+        args['_callbackId'] = callbackId;
+      }
+      args = JSON.stringify(args || {});
+      console.log('===callHandler', name, args, callback);
 
-        function callSyn() {
-           alert(dsBridge.call("testSyn", {msg: "testSyn"}))
-        }
+      // WKWebview
+      if (internalBridge.wk) {
+        ret = prompt(internalBridgeName + '=' + name, args);
+      } else {
+        ret = internalBridge.callHandler(name, args);
+      }
+      return ret;
+    },
+    registerHandler: function (name, handler) {
+      console.log('===registerHandler' + name);
+      this.handlerMap[name] = handler;
+    },
+  };
 
-        function callAsyn() {
-            dsBridge.call("testAsyn", {msg: "testAsyn"}, function (v) {
-                  alert(v)
-            })
+  // extend internal bridge
+  internalBridge.invokeHandler = function (name, args, callbackId) {
+    console.log('===invokeHandler', name, args);
+    var handler = jsBridge.handlerMap[name];
+    if (handler.length === 2) {
+      // async handler
+      handler(args, function (ret) {
+        if (typeof callbackId === 'undefined') return;
+        if (internalBridge.wk) {
+          prompt(internalBridgeName + 'cid=' + callbackId, JSON.stringify(ret));
+        } else {
+          internalBridge.returnValue(callbackId, JSON.stringify(ret));
         }
+      });
+    } else {
+      internalBridge.returnValue(callbackId, JSON.stringify(handler(args)));
+    }
+  };
+  internalBridge.invokeCallback = function (callbackId, result, complete) {
+    console.log('===invokeCallback' + complete);
+    console.log(callbackId);
+    console.log(result);
+    console.log(typeof complete);
+    var callback = jsBridge.callbackMap[callbackId];
+    if (complete) {
+      delete jsBridge.callbackMap[callbackId];
+    }
+    callback(result);
+  };
 
-        function callAsyn_() {
-            for(var i=0;i<2000;i++){
-                dsBridge.call("testAsyn", {msg: "js+"+i}, function (v) {
-                    if(v=="js+1999 [ asyn call]"){
-                        alert("全部任务执行结束")
-                    }
-                 })
-            }
-        }
+  return jsBridge;
+}
+/* BRIDGE END */
 
-        function callNoArgSyn() {
-            alert(dsBridge.call("testNoArgSyn"));
-        }
+// bridge demo
 
-        function callNoArgAsyn() {
-            dsBridge.call("testNoArgAsyn", function (v) {
-                alert(v)
-            });
-        }
+var bridge = getJsBridge();
+console.log(bridge);
 
-        function callNever() {
-            dsBridge.call("testNever", {msg: "testSyn"})
-        }
+function callSync() {
+  alert(bridge.callHandler('testSync', { msg: 'testSync' }));
+}
 
-        function callProgress(){
-         dsBridge.call("callProgress", function (value) {
-                document.getElementById("progress").innerText=value
-          })
-        }
+function callAsync() {
+  alert(bridge.callHandler('testAsync', { msg: 'testAsync' }, function (ret) {
```

Comment:
can you delete the outter alert() method, it makes the call sync if you use alert().
you can just call callHandler() method

Reply:
sure.